api/docker/jobservice.go

```diff
@@ -0,0 +1,89 @@
+package docker
+
+import (
+	"bytes"
+	"context"
+	"strconv"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/api/types/strslice"
+	"github.com/portainer/portainer"
+	"github.com/portainer/portainer/archive"
+)
+
+type JobService struct {
+	DockerClientFactory *ClientFactory
+}
+
+func NewJobService(dockerClientFactory *ClientFactory) *JobService {
+	return &JobService{
+		DockerClientFactory: dockerClientFactory,
+	}
+}
+
+func (service *JobService) Execute(endpoint *portainer.Endpoint, image string, script []byte) error {
+	buffer, createFileErr := archive.TarFileInBuffer(script, "script.sh", 0700)
+	if createFileErr != nil {
+		return createFileErr
+	}
+
+	cli, createClientErr := service.DockerClientFactory.CreateClient(endpoint)
+	if createClientErr != nil {
+		return createClientErr
+	}
+	defer cli.Close()
+
+	// containerName := "test"
+
+	// TODO this will run only bash scripts
+	cmd := make([]string, 2)
+	cmd[0] = "sh"
+	cmd[1] = "/tmp/script.sh"
+
+	_, imagePullErr := cli.ImagePull(context.Background(), image, types.ImagePullOptions{})
```

Comment:
Somehow this triggers a background job that will pull the image. Meaning that the image will probably not be available during the container creation step.

Can you try to figure out if we can wait for the image to be ready? 

Also, docs state that this function returns a `io.ReadCloser` that needs to be close.